packages/builder/src/helpers/validation/yup/app.js

```diff
@@ -0,0 +1,80 @@
+import { string, mixed } from "yup"
+import { APP_NAME_REGEX, APP_URL_REGEX } from "constants"
+
+export const name = (validation, { apps, currentApp } = { apps: [] }) => {
+  validation.addValidator(
+    "name",
+    string()
+      .trim()
+      .required("Your application must have a name")
+      .matches(APP_NAME_REGEX, "App name must be letters and numbers only")
+      .test(
+        "non-existing-app-name",
+        "Another app with the same name already exists",
+        value => {
+          if (!value) {
+            // exit early, above validator will fail
+            return true
+          }
+          if (currentApp) {
+            // filter out the current app if present
+            apps = apps.filter(app => app.appId !== currentApp.appId)
+          }
+          return !apps
+            .map(app => app.name)
+            .some(appName => appName.toLowerCase() === value.toLowerCase())
+        }
+      )
+  )
+}
+
+export const url = (validation, { apps, currentApp } = { apps: [] }) => {
+  validation.addValidator(
+    "url",
+    string()
+      .nullable()
+      .matches(APP_URL_REGEX, "App URL must not contain spaces")
+      .test(
+        "non-existing-app-url",
+        "Another app with the same URL already exists",
+        value => {
+          // url is nullable
+          if (!value) {
+            return true
+          }
+          if (currentApp) {
+            // filter out the current app if present
+            apps = apps.filter(app => app.appId !== currentApp.appId)
+          }
+          return !apps
+            .map(app => app.url)
+            .some(appUrl => appUrl.toLowerCase() === value.toLowerCase())
```

Comment:
Thanks for pointing this out @aptkingston! I was under the assumption the url would always be set. @mike12345567 could you confirm for us, are we always supposed to have an app url, even for historical apps or do we need to account for a point when they didn't have them?

Reply:
The URL should be set for all apps, as it was being set to the app name based URL before this update - there would have been a point in time where this wasn't true but it probably 6-8 months ago, not sure apps from then would even work without being re-imported (which would generate a URL for them).

Reply:
Thanks! sounds like it might affect some older self hosted installations then. I'll account for it not being set in this case, and look into a migration that will set the url if it doesn't exist already